src/pages/home/report/ReportActionsList.js

```diff
@@ -227,14 +229,28 @@ function ReportActionsList({
             unreadActionSubscription.current.remove();
             unreadActionSubscription.current = null;
         }
+        if (readNewestActionSubscription.current) {
+            readNewestActionSubscription.current.remove();
+            readNewestActionSubscription.current = null;
+        }
 
-        // Listen to specific reportID for unread event and set the marker to new message
-        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+        const resetUnreadMarker = (newLastReadTime) => {
             cacheUnreadMarkers.delete(report.reportID);
             lastReadTimeRef.current = newLastReadTime;
             setCurrentUnreadMarker(null);
+        };
+
+        // Listen to specific reportID for unread event and set the marker to new message
+        unreadActionSubscription.current = DeviceEventEmitter.addListener(`unreadAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
             setMessageManuallyMarkedUnread(new Date().getTime());
         });
+
+        // Listen to specific reportID for read newest action event and reset the marker
+        readNewestActionSubscription.current = DeviceEventEmitter.addListener(`readNewestAction_${report.reportID}`, (newLastReadTime) => {
+            resetUnreadMarker(newLastReadTime);
+            setMessageManuallyMarkedUnread(0);
+        });
```

Comment:
- Make `ReportActionsList` a directory with `index.js` file
- Add a file `DeviceEventListenerRef.ts` in the new directory:

```ts
class DeviceEventListenerRef {
    // ...
    
    static use(
        type: string,
        listener: (data: unknown) => void
    ): DeviceEventListenerRef {
        // ...
    }

    add(): void {
        // ...
    }

    remove(): void {
        // ...
    }
}

```